### DISCUSSION TITLE
Want to acces 2 components for my community demo game: cannot borrow `world` as mutable because it is also borrowed as immutable

### DISCUSSION THREAD

**Discussion by Retrodad0001** [bevyengine/bevy#13769] [Q&A] [answered]
I'm new to rust and bevy, so it could be that I'm missing something.

Public repo:  https://github.com/Retrodad0001/evil_dungeon

I want to create a simple but polished game for the community to learn rust, bevy, and the "ECS way" of designing a game and share that knowledge with everyone.

error[E0502]: cannot borrow `world` as mutable because it is also borrowed as immutable
   --> src\core\screen_playing.rs:289:52
    |
285 |         let entity_a_world_ref: EntityRef = world.entity(entity_a);
    |                                             ----- immutable borrow occurs here
...
289 |         let mut entity_world_mut: EntityWorldMut = world.entity_mut(entity_b);
    |                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
...
293 |         if component_damage_option.is_none() || component_health_option.is_none() {
    |            ----------------------- immutable borrow later used here

```
pub(crate) fn collision_event_handle_damage_dealing_and_health_for_all(
    mut event_collision_detected: EventReader<EventCollisionDetected>,
    mut event_actor_is_killed: EventWriter<EventActorIsKilled>,
    mut world: World,
) {
    for collision_event in event_collision_detected.read() {
        let entity_a: Entity = collision_event.entity_a;
        let entity_b: Entity = collision_event.entity_b;

        let entity_a_world_ref: EntityRef = world.entity(entity_a);
        let component_damage_option =
            entity_a_world_ref.get::<ComponentCanDealDamage>();

        let mut entity_world_mut: EntityWorldMut = world.entity_mut(entity_b);
        let component_health_option =
            entity_world_mut.get_mut::<ComponentHasHealth>();

        if component_damage_option.is_none() || component_health_option.is_none() {
            continue;
        }

        let mut health  = component_health_option.unwrap();
        let damage   = component_damage_option.unwrap();

        health.current_health -= damage.damage_amount;

        if health.current_health <= 0 {
            event_actor_is_killed.send(EventActorIsKilled::new(entity_b));
        }
    }
}
```

**Comment by MrGVSV** [comment, marked as answer]
Is `collision_event_handle_damage_dealing_and_health_for_all` a system? If so, I’d recommend using a `Query` instead of `World` so that you can use something like [`Query::get_many_mut`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#method.get_many_mut) to fetch both entities without breaking Rust's mutability rules.
